### PR TITLE
Add population type and dimensions

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

[Trello card](https://trello.com/c/h2wdw2YX/1186-update-search-data-importer-to-include-population-type-dimensions-in-transformation-and-map-to-elasticsearch-doc)

- Add population types and dimensions to elasticsearch and kafka structs and schemas.
- Add Dockerfile.local and reflex to make this repo available for dp-compose/v2
- Upgrade dp-kafka dependency to version 3
- Implement component tests (including `dimensions` and `population_type` validation)

### How to review

- Make sure code changes make sense
- Make sure unit tests pass
- Make sure component tests pass

### Who can review

Anyone

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -15,23 +15,32 @@ type Config struct {
...
 	AwsRegion                  string        `envconfig:"AWS_REGION"`
 	AwsService                 string        `envconfig:"AWS_SERVICE"`
 	SignElasticsearchRequests  bool          `envconfig:"SIGN_ELASTICSEARCH_REQUESTS"`
+	StopConsumingOnUnhealthy   bool          `envconfig:"STOP_CONSUMING_ON_UNHEALTHY"`
+	Kafka                      *Kafka
+}
+
+// Kafka contains the config required to connect to Kafka
+type Kafka struct {
+	PublishedContentGroup     string   `envconfig:"KAFKA_PUBLISHED_CONTENT_GROUP"`
```
Comment (nshumoogum) [inline config/config.go:30]: Is it worth ordering these alphabetically?

File: config/config.go [hunk 2]
```diff
@@ -48,23 +57,29 @@ func Get() (*Config, error) {
...
-		PublishedContentTopic:      "search-data-import",
 		BatchSize:                  500,
 		BatchWaitTime:              time.Second * 5,
 		ElasticSearchAPIURL:        "http://localhost:11200",
 		AwsRegion:                  "eu-west-1",
 		AwsService:                 "es",
 		SignElasticsearchRequests:  false,
+		StopConsumingOnUnhealthy:   true,
+		Kafka: &Kafka{
+			PublishedContentGroup:     "dp-search-data-importer",
```
Comment (nshumoogum) [inline config/config.go:68]: Same as last comment

File: config/config_test.go [hunk 3]
```diff
@@ -24,22 +24,27 @@ func TestConfig(t *testing.T) {
...
-				So(cfg.PublishedContentGroup, ShouldEqual, "dp-search-data-importer")
-				So(cfg.PublishedContentTopic, ShouldEqual, "search-data-import")
 				So(cfg.BatchSize, ShouldEqual, 500)
 				So(cfg.BatchWaitTime, ShouldEqual, time.Second*5)
 				So(cfg.ElasticSearchAPIURL, ShouldEqual, "http://localhost:11200")
 				So(cfg.AwsRegion, ShouldEqual, "eu-west-1")
 				So(cfg.AwsService, ShouldEqual, "es")
 				So(cfg.SignElasticsearchRequests, ShouldEqual, false)
+				So(cfg.StopConsumingOnUnhealthy, ShouldBeTrue)
+				So(cfg.Kafka.PublishedContentGroup, ShouldEqual, "dp-search-data-importer")
```
Comment (nshumoogum) [inline config/config_test.go:34]: Same again, order assertions alphabetically - easier to find whether we have an assertion or not for a configured variable

File: features/steps/assertor.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+)
+
+// ElasticSearchCall represents a call to ElasticSearch, containing
```
Comment (nshumoogum) [inline features/steps/assertor.go:12]: fyi Elasticsearch is one word - the database is called Elasticsearch and it is Elastic companies search engine

File: models/event.go [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
 
-// SearchDataImportModel provides an avro structure for a SearchDataImportModel Called event
-type SearchDataImportModel struct {
+// SearchDataImport provides an avro structure for a SearchDataImport Called event
+type SearchDataImport struct {
 	UID             string               `avro:"uid"`
 	URI             string               `avro:"uri"`
 	Edition         string               `avro:"edition"`
-	DataType        string               `avro:"type"`
+	DataType        string               `avro:"data_type"`
```
Comment (nshumoogum) [inline models/event.go:8]: Why is this changing from `type` to` data_type`?

Comment (nshumoogum) [reply models/event.go:8]: Was this previously broken?

Comment (nshumoogum) [reply models/event.go:8]: I can see that the extractor sets this to `data_type` too - https://github.com/ONSdigital/dp-search-data-extractor/blob/develop/schema/schema.go#L31

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 1 above)
Comment (DavidSubiros) [reply config/config.go:30]: Ok, I will do that.

File: features/steps/assertor.go (hunk 4 above)
Comment (DavidSubiros) [reply features/steps/assertor.go:12]: thanks, I will rename it

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**